sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -2541,6 +2541,11 @@ class DataFrameSuite extends QueryTest
       assert(df.schema === new StructType().add(StructField("d", DecimalType(38, 0))))
     }
   }
+
+  test("SPARK-32640: ln(NaN) should return NaN") {
+    val df = Seq(Double.NaN).toDF("d")
+    checkAnswer(df.selectExpr("ln(d)"), Row(Double.NaN))
```

Comment:
Question: does this actually trigger codegen? It gets converted to LocalRelation during logical optimization, right?
(If this test fails before reverting Janino and passes after the revert, then I guess it's still valid to have it here, otherwise I'm a bit confused)

Reply:
Yea, it does; `ConvertToLocalRelation` disabled in tests: https://github.com/apache/spark/blob/8b119f1663dc65c96f20606ac8405169d8f9d31d/sql/core/src/test/scala/org/apache/spark/sql/test/SharedSparkSession.scala#L70-L74

Reply:
Cool, thanks for confirming!